doc/man3/SSL_CTX_set1_curves.pod

```diff
@@ -174,6 +174,33 @@ and the client supports C<P-384> including key share for this group, a HRR will
 triggered for C<P-521> despite the availability of a key share for P-384, which
 overlaps with a lower priority server-side tuple.
 
+For example, consider the same server but different client with I<list>
+"P-521:P-384:*ffdhe2048". When the client attempts connecting to the server,
+only a keyshare for ffdhe2048 will be sent. But because P-521 and P-384 are
+supported by the client, an HRR will be triggered for P-521 (preferred by the
+server).
+
+As another example, consider a server I<list> "A:B/C:D/E:F". Listed in order of
+highest preference to least, 3 group tuples are created: "A:B", "C:D", and "E:F".
+Some example scenarios for the client I<list>:
+
+- "A:D:*F": The server prefers "A" but did not receive a key share for "A", so
+  an HRR is triggered for "A".
+
+- "B:*C": The server prefers "B" from the first group tuple "A:B", so an HRR is
+  triggered for "B".
+
+- "C:*F": The server prefers "C" from the second group tuple "C:D", so an HRR is
+  triggered for "C".
+
+- "C:*D": Even though the server prefers "C" over "D", the server will accept
+  the key share for "D". Within a tuple, existing keyshares trump preference
+  order.
+
+- "*C:*D": The server accepts the "C" key share.
+
+- "F": The server accepts "F" key share.
```

Comment:
Shouldn't we be explicit in these examples about whether we are using client preference or server preference - and include some examples of how things might turn out differently depending on which one you choose.

Reply:
I've moved the examples to its own section and specified differences between client and server preference. Please help me check that all the examples are correct